File: tests/fixtures/steps-validator/python/features/steps/bad_steps.py
"""
Bad step definitions without proper assertions.
Used for testing the steps validator.
"""
from behave import given, when, then

result = None


@given("a bad setup")
def step_given_bad_setup(context):
    # TODO: implement later
    pass


@when("bad action happens")
def step_when_bad_action(context):
    print("Doing something...")


# ❌ BAD: Only print, no assertion!
@then("the result is verified")
def step_then_result_verified(context):
    print("[BadSteps] ✅ Result verified")
    print("[BadSteps] All good!")


# ❌ BAD: Only pass
@then("the operation completes")
def step_then_operation_completes(context):
    pass


# ❌ BAD: Only return
@then("the data is processed")
def step_then_data_processed(context):
    return


# ❌ BAD: NotImplementedError
@then("the validation passes")
def step_then_validation_passes(context):
    raise NotImplementedError()


# ⚠️ WARNING: TODO comment
@then("the feature works")
def step_then_feature_works(context):
    # TODO: add proper assertion
    print("Feature working...")


# ⚠️ WARNING: print only in Then
@then("the log is written")
def step_then_log_written(context):
    print("[BadSteps] Log written")

File: tests/fixtures/steps-validator/typescript/steps/bad.steps.ts
import { Given, When, Then } from "@cucumber/cucumber";

let result: string;

Given("a bad setup", async function () {
  // TODO: implement later
});

When("bad action happens", async function () {
  console.log("Doing something...");
});

// ❌ BAD: Only console.log, no assertion!
Then("the result is verified", async function () {
  console.log("[BadSteps] ✅ Result verified");
  console.log("[BadSteps] All good!");
});

// ❌ BAD: Empty body
Then("the operation completes", async function () {});

// ❌ BAD: Only return
Then("the data is processed", async function () {
  return;
});

// ❌ BAD: Pending error
Then("the validation passes", async function () {
  throw new Error("Pending: not implemented");
});

// ⚠️ WARNING: TODO comment
Then("the feature works", async function () {
  // TODO: add proper assertion
  console.log("Feature working...");
});

// ⚠️ WARNING: console.log only in Then
Then("the log is written", async function () {
  console.log("[BadSteps] Log written");
});
